controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1131,6 +1137,363 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
+            *ap_metrics_tlv->estimated_service_info_field_ac_bk(i) =
+                metric_data_per_agent.estimated_service_info_field_ac_bk[i];
+            *ap_metrics_tlv->estimated_service_info_field_ac_vo(i) =
+                metric_data_per_agent.estimated_service_info_field_ac_vo[i];
+            *ap_metrics_tlv->estimated_service_info_field_ac_vi(i) =
+                metric_data_per_agent.estimated_service_info_field_ac_vi[i];
+        }
+    }
+
+    if (database.setting_certification_mode()) {
+        auto certification_tx_buffer = database.get_certification_tx_buffer();
+        if (!certification_tx_buffer) {
+            LOG(ERROR) << "certification_tx_buffer is not allocated!";
+            return false;
+        }
+        database.fill_certification_tx_buffer(cmdu_tx);
+        return true;
+    }
+    return son_actions::send_cmdu_to_agent(sd, cmdu_tx);
+}
+
+bool master_thread::handle_cmdu_1905_ap_metric_response(Socket *sd,
+                                                        ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received AP_METRIC_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
+
+    //getting reference for ap metric data storage from db
+    auto &ap_metric_data = database.get_ap_metric_data_map();
+
+    //will hold new ap metric data from Reporting Agent
+    // son::node::ap_metrics_data new_ap_metric_data;
+
+    int tlvType;
+    // holds mac address of the reporting agent, used as ap_metric_data key
+    // sMacAddr reporting_agent_bssid;
+
+    // parse all tlvs in cmdu and save ap metric data to database
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
+
+        if (tlvType < 0) {
+            LOG(ERROR) << "getNextTlvType has failed";
+            return false;
+        }
+
+        if (tlvType == int(wfa_map::eTlvTypeMap::TLV_AP_METRIC)) {
+            //parse tx_ap_metric_data
+            auto ap_metric_tlv = cmdu_rx.addClass<wfa_map::tlvApMetric>();
+            if (!ap_metric_tlv) {
+                LOG(ERROR) << "addClass wfa_map::tlvApMetric has failed";
+                return false;
+            }
+
+            sMacAddr reporting_agent_bssid = ap_metric_tlv->bssid();
+
+            LOG(DEBUG) << "recieved tlvApMetric from BSSID ="
+                       << network_utils::mac_to_string(reporting_agent_bssid);
+
+            //fill tx data from TLV
+            if (!ap_metric_data[reporting_agent_bssid].add_ap_metric_data(ap_metric_tlv)) {
+                LOG(ERROR) << "adding apMetricData from tlv has failed";
+                return false;
+            }
+
+            //save agent ap metric data to database or modify already existing.
+
+            std::string reporter_status_s = "new";
+            auto reporting_map_it         = ap_metric_data.find(reporting_agent_bssid);
+
+            if (reporting_map_it != ap_metric_data.end()) {
```

Comment:
This makes no sense... You just added it a few lines above, so it will never be "new"! That shows that this piece of code that checks if it is new or not is completely useless.